src/peer_sourcehash.cc

```diff
@@ -137,28 +137,25 @@ peerSourceHashRegisterWithCacheManager(void)
 CachePeer *
 peerSourceHashSelectParent(PeerSelector *ps)
 {
-    const char *c;
     CachePeer *p = nullptr;
     unsigned int user_hash = 0;
     unsigned int combined_hash;
     double score;
     double high_score = 0;
-    const char *key = nullptr;
-    char ntoabuf[MAX_IPSTRLEN];
 
     if (SourceHashPeers().empty())
         return nullptr;
 
     assert(ps);
     HttpRequest *request = ps->request;
 
-    key = request->client_addr.toStr(ntoabuf, sizeof(ntoabuf));
+    const auto key = request->client_addr.toStrAsSBuf();
```

Comment:
This use-case does not need any new API. It can just:
```suggestion
    const auto key = ToSBuf(request->client_addr);
```

Reply:
Do we want to guarantee that two different Squid instances produce the same hash for the same request?

* If yes, then we may want to avoid changing format/hash to avoid  breaking that guarantee for instances running different Squid versions. We could still change if there is a compelling reason to do so, of course, but I do not see such a reason here.
* Otherwise, we should not covert the IP address to text to compute a hash.

Reply:
Fair point.
I think we should try to preserve behaviour in a major version,
but I wouldn't feel constrained to do the same across
major versions.

At the same time, I would keep this change as its own PR

Reply:
> This use-case does not need any new API. It can just:

Adopted

Reply:
> I wouldn't feel constrained to do the same across major versions.

Is this hash computation (de facto) standardized across different products (including non-Squid-based caches) or can we assume that this is a Squid-specific concern?

If it is a Squid-specific concern: The "we can change things across versions" logic is sound, but it does not work as well when dealing with distributed cache hierarchies where nodes may run different versions. I would not object to a breaking change like that if others are sure that it is warranted, but in that case, again, we should not covert the IP address to text to compute a hash.

The non-obvious sensitivity of this code to address formatting warrants a C++ comment, especially if we migrate to something implicit like ToSBuf().

> I would keep this change as its own PR

Thank you. I assume "this change" is "possible change of peer hash values".